### PR TITLE
storage_interface refactor to use span and references

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: examples/connection_tester.cpp [hunk 1]
```diff
@@ -846,7 +846,7 @@ void generate_data(char const* path, torrent_info const& ti)
 			generate_block(piece, i, j, 0x4000);
 			file::iovec_t b = { piece, 0x4000};
 			storage_error error;
-			st->writev(&b, 1, i, j, 0, error);
+			st->writev({b}, i, j, 0, error);
```
Comment (arvidn) [inline examples/connection_tester.cpp]: in fact, I think you don't even need the braces

File: src/disk_io_thread.cpp [hunk 2]
```diff
@@ -616,9 +616,9 @@ namespace libtorrent
 		bool failed = false;
 		for (int i = 1; i <= num_blocks; ++i)
 		{
-			if (i < num_blocks && flushing[i] == flushing[i-1]+1) continue;
-			int ret = pe->storage->get_storage_impl()->writev(iov_start
-				, i - flushing_start
+			if (i < num_blocks && flushing[i] == flushing[i - 1] + 1) continue;
+			int ret = pe->storage->get_storage_impl()->writev({iov_start
+					, size_t(i - flushing_start)}
```
Comment (arvidn) [inline src/disk_io_thread.cpp]: if you break the line befor { you don't need the additional indentation level

File: examples/connection_tester.cpp (hunk 1 above)
Comment (aldenml) [inline examples/connection_tester.cpp]: yes, I was thinking about it right now...I will change it in all places....

File: test/test_storage.cpp [hunk 3]
```diff
@@ -232,58 +232,58 @@ void run_storage_tests(std::shared_ptr<torrent_info> info
 
 	// write piece 1 (in slot 0)
 	file::iovec_t iov = { piece1.data(), half};
-	ret = s->writev(&iov, 1, 0, 0, 0, ec);
+	ret = s->writev({iov}, 0, 0, 0, ec);
```
Comment (arvidn) [inline test/test_storage.cpp]: same here, I don't think you need the braces

File: test/test_transfer.cpp [hunk 4]
```diff
@@ -104,10 +103,10 @@ struct test_storage : default_storage
 			return 0;
 		}
 
-		for (int i = 0; i < num_bufs; ++i)
+		for (int i = 0; i < int(bufs.size()); ++i)
```
Comment (arvidn) [inline test/test_transfer.cpp]: I think you could just do: for (auto const& b : bufs) m_written += int(b.iov_len);
here

**Comment by aldenml** [comment]
do you mind cancel my unnecessary travis jobs

**Pull request merged**